src/Cardano/Prelude/Show.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}   -- for Rep ForgettingFields
+
+-- | This module uses "GHC.Generics" to define 'gshowsPrecForgetFields', which
+-- can often make the result of 'show' more legible, especially for @newtype@s
+-- for example.
+--
+module Cardano.Prelude.Show (
+    gshowsPrecForgetFields,
+    GShowK1 (..),
+  ) where
+
+import           Prelude
+
+import           Data.Coerce
+import           Data.Kind (Type)
+import           GHC.Generics
+
+import qualified Generics.Deriving.Show as GD
+
+-- | Implements 'showsPrec' as if the constructors' record fields were instead
+-- plain arguments
+--
+gshowsPrecForgetFields ::
```

Comment:
Hmm, I feel like I might be confused when seeing this used in other projects. I'd probably would wonder (as I did), "what does it mean to forget fields?".

After reading through this implementation, I understand that the aim of this is to render a value of some constructor defined using record syntax as if it were defined with plain arguments. However, I'm not sure that I gather that from the name, `*ForgetFields`.

@erikd: Any thoughts on this?

Reply:
Naming things is difficult, especially for whoever is writing the code. Maybe `*DropFieldNames` ?

Reply:
* First word one of: `Forget`, `Drop`, `Without`, `Suppress`
* Second word one of: `Fields`, `FieldNames`, `Records`, `RecordSyntax`

I think I favor `gshowsPrecWithoutRecordSyntax`. Thoughts?

Reply:
I still prefer `gshowsPrecDropFieldNames` but that may be because I came up with it.

Maybe `gshowsPrecSimpleRecordSyntax` ?

Reply:
I hesitate on `gshowsPrecSimpleRecordSyntax` because the function eliminates the record syntax -- it's not _simple_ but rather _gone_ :)

`gshowsPrecDropFieldNames` seems correct to me, though I also favor my own child `gshowsPrecWithoutRecordSyntax` :)

Reply:
A vote for `gshowsPrecWithoutRecordSyntax`

Reply:
Lots of good choices :smile: I think I'd be cool with any of:

- `*DropFieldNames`
- `*WithoutFieldNames`
- `*WithoutRecordSyntax`

I'm leaning more toward something with `*FieldNames`, but I think that any of them are clearer than `*ForgetFields`.

Reply:
I went with `WithoutRecordSyntax`, it got votes from me and Thomas and a nod from Luke. Erik, is it acceptable?

I also exported that type family since it occurs in the type of an exported function.

I think it's ready for review again. Thanks, @erikd!